packages/rpc-core/src/rpc-methods/__tests__/simulate-transaction-test.ts

```diff
@@ -0,0 +1,717 @@
+
+    it('throws when called with a malformed transaction message', async () => {
+        expect.assertions(1);
+        const secretKey = await getSecretKey();
+        const message = new Uint8Array([4, 5, 6]);
+        const signature = new Uint8Array(await crypto.subtle.sign('Ed25519', secretKey, message));
+        const resultPromise = rpc
+            .simulateTransaction(
+                Buffer.from(
+                    new Uint8Array([
+                        0x01, // Length of signatures
+                        ...signature,
+                        ...message,
+                    ])
+                ).toString('base64') as Base64EncodedWireTransaction,
+                { commitment: 'processed', encoding: 'base64' }
+            )
+            .send();
+
+        await expect(resultPromise).rejects.toMatchObject({
+            code: -32602 satisfies (typeof SolanaJsonRpcErrorCode)['JSON_RPC_INVALID_PARAMS'],
+            message: expect.stringContaining('failed to fill whole buffer'),
+            name: 'SolanaJsonRpcError',
+        });
+    });
+
+    it('returns an AccountNotFound error when the fee payer is an unknown account', async () => {
+        expect.assertions(1);
+        const [[secretKey, publicKeyBytes], { value: latestBlockhash }] = await Promise.all([
+            (async () => {
+                const keyPair = (await crypto.subtle.generateKey('Ed25519', /* extractable */ false, [
+                    'sign',
+                    'verify',
+                ])) as CryptoKeyPair;
+                return [keyPair.privateKey, new Uint8Array(await crypto.subtle.exportKey('raw', keyPair.publicKey))];
+            })(),
+            rpc.getLatestBlockhash().send(),
+        ]);
+        const message = getMockTransactionMessage({
+            blockhash: latestBlockhash.blockhash,
+            feePayerAddressBytes: publicKeyBytes,
+            memoString: `Hello from the web3.js tests! [${performance.now()}]`,
+        });
+        const signature = new Uint8Array(await crypto.subtle.sign('Ed25519', secretKey, message));
+        const resultPromise = rpc
+            .simulateTransaction(
+                Buffer.from(
+                    new Uint8Array([
+                        0x01, // Length of signatures
+                        ...signature,
+                        ...message,
+                    ])
+                ).toString('base64') as Base64EncodedWireTransaction,
+                { commitment: 'processed', encoding: 'base64' }
+            )
+            .send();
+
+        await expect(resultPromise).resolves.toMatchObject({
+            value: expect.objectContaining({
+                accounts: null,
+                err: 'AccountNotFound',
+                logs: expect.any(Array),
+                returnData: null,
+                unitsConsumed: expect.any(BigInt),
+            }),
+        });
+    });
+
+    it('returns account data for a transaction account with base64 encoding', async () => {
```

Comment:
Just curious - bc the docs aren't super clear - this returns the account data post-simulation? In other words, it returns the simulated change in state for any provided keys? Pretty cool if true, didn't know that

Reply:
I haven't used this API much but IIRC yep it's the account after the transaction. I'm pretty sure I used this in a fake-wallet to show the change in SOL that would result from approving a transaction, by comparing lamports with a `getAccountInfo` or similar call against the same account.

Reply:
Interesting!